### PR TITLE
nmcli: Check dns4 param for None

### LINKED ISSUE #34012: nmcli module fails with self.dns4=' '.join(module.params['dns4']) TypeError
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

`nmcli`

##### ANSIBLE VERSION

```
ansible 2.4.1.0
  config file = /Users/dlbewley/src/ansible/playbook-openshift/ansible.cfg
  configured module search path = [u'/Users/dlbewley/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.4.1.0/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

##### OS / ENVIRONMENT

- Manager: OS X
- Managed: Red Hat Enterprise Linux Server release 7.4 (Maipo)

##### SUMMARY

Playbook fails when trying to join `None` value for `dns4` param [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/nmcli.py#L559)

I do not see a requirement to include dns servers, and expect to use DHCP.

##### STEPS TO REPRODUCE

Host with links on eno1, eno2. Int eno1 is def gw

```yaml
---
- hosts: bonded

# Dec 18 18:08:43 ose-prod-node-07 ansible-nmcli[31031]: Invoked with conn_name=cluster ingress=None slavepriority=32 vlandev=None forwarddelay=15 egress=None ageingtime=300 mtu=None hellotime=2 maxage=20 vlanid=None priority=128 gw4=None state=present gw6=None master=None stp=True ifname=None type=bond miimon=None arp_ip_target=None downdelay=None mac=None ip6=None ip4=None autoconnect=None dns6=None dns4=None arp_interval=None flags=None mode=802.3ad updelay=None

  vars:
    nmcli_bond:
      - conn_name: cluster
        mode: 802.3ad
        mtu: 9000

    nmcli_bond_slave:
      - conn_name: eno1
        master: cluster
      - conn_name: eno2
        master: cluster

  tasks:
    - name: create bond
      nmcli:
        type: bond
        conn_name: '{{ item.conn_name }}'
        mode: '{{ item.mode }}'
        state: present
      with_items:
        - '{{ nmcli_bond }}'

    - name: add interfaces to bond
      nmcli:
        type: bond-slave
        conn_name: '{{ item.conn_name }}'
        ifname: '{{ item.ifname }}'
        master: '{{ item.master }}'
        state: present
      with_items:
        - '{{ nmcli_bond_slave }}'
```

##### EXPECTED RESULTS

First test, but expect playbook to run without error.

##### ACTUAL RESULTS

```
failed: [ose-prod-node-07.example.com] (item={u'conn_name': u'cluster', u'mode': u'802.3ad', u'mtu': 9000}) => {
    "changed": false,
    "failed": true,
    "item": {
        "conn_name": "cluster",
        "mode": "802.3ad",
        "mtu": 9000
    },
    "module_stderr": "OpenSSH_7.4p1, LibreSSL 2.5.0\r\ndebug1: Reading configuration data /Users/dlbewley/.ssh/config\r\ndebug1: /Users/dlbewley/.ssh/config line 3: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 51: Applying options for *\r\ndebug1: /etc/ssh/ssh_config line 56: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 10219\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to ose-prod-node-07.example.com closed.\r\n",
    "module_stdout": "/tmp/ansible_gKn2an/ansible_module_nmcli.py:493: PyGIWarning: NetworkManager was imported without specifying a version first. Use gi.require_version('NetworkManager', '1.0') before import to ensure that the right version gets loaded.\r\n  from gi.repository import NetworkManager, NMClient\r\n/tmp/ansible_gKn2an/ansible_module_nmcli.py:493: PyGIWarning: NMClient was imported without specifying a version first. Use gi.require_version('NMClient', '1.0') before import to ensure that the right version gets loaded.\r\n  from gi.repository import NetworkManager, NMClient\r\nTraceback (most recent call last):\r\n  File \"/tmp/ansible_gKn2an/ansible_module_nmcli.py\", line 1190, in <module>\r\n    main()\r\n  File \"/tmp/ansible_gKn2an/ansible_module_nmcli.py\", line 1134, in main\r\n    nmcli=Nmcli(module)\r\n  File \"/tmp/ansible_gKn2an/ansible_module_nmcli.py\", line 559, in __init__\r\n    self.dns4=' '.join(module.params['dns4'])\r\nTypeError\r\n",
    "msg": "MODULE FAILURE",
    "rc": 1
}
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix check dns4 parameter in nmcli module for None value.

Fixes: #34012

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/modules/net_tools/nmcli.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by mkrizek** [comment]
Also fixes https://github.com/ansible/ansible/issues/31737 which has already PR open https://github.com/ansible/ansible/pull/30757 but it looks like the contributor disappeared?

File: lib/ansible/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -556,7 +556,7 @@ def __init__(self, module):
         self.type = module.params['type']
         self.ip4 = module.params['ip4']
         self.gw4 = module.params['gw4']
-        self.dns4 = ' '.join(module.params['dns4'])
+        self.dns4 = ' '.join(module.params['dns4']) if module.params.get('dns4') else None
```
Comment (mkrizek) [inline lib/ansible/modules/net_tools/nmcli.py:559]: Looks good to me. Maybe add a test?

Comment (Akasurde) [reply lib/ansible/modules/net_tools/nmcli.py:559]: There is no existing test for nmcli and I am not aware of all the possible scenarios of nmcli, so may not be the right person to add new tests.

Comment (mkrizek) [reply lib/ansible/modules/net_tools/nmcli.py:559]: No need to add other scenarios, it's fine to add a test for just this one use case, but agreed that `nmcli` might be tricky to test.

**Review by gundalow — APPROVED** [review]

**Comment by Akasurde** [comment]
@mattclay Could you please help me in understanding this CI error, it is working in my current environment?

**Comment by Akasurde** [comment]
Closing in-favor of #30757.

**Pull request closed without merging**